base/sparse/linalg.jl

```diff
@@ -532,10 +531,13 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
     end
     ind = Array(Int64, n)
     ind_hist = Array(Int64, maxiter * t)
-    S = zeros(T, n, t)
+
+    Ti = T <: Real ? Float64 : Complex128
```

Comment:
Maybe I overlooked something, but it seems to me that because of `est = zero(real(eltype(Y)))` we have two  issues:
* a complex value for a norm
* a double precision norm (estimate) for a single precision matrix
I think that the issue was already there and depended on the type declaration `est = zero(real(eltype(Y)))`, but I'm not sure what the best way to fix it may be.